app/models/tagging.rb

```diff
@@ -46,9 +46,11 @@ def update_taggings_count
   end
 
   def update_search
-    reindex_boundary = 100
-    if %w(Character Relationship Freeform).include?(tagger.type)
-      tagger.enqueue_to_index if tagger.taggings_count < reindex_boundary
-    end
+    # When a tag's count is more than this threshold, we start caching it,
+    # so taggings_count becomes less accurate. That's our cue to stop
+    # eagerly reindexing counts as well.
+    return unless %w[Fandom Character Relationship Freeform].include?(tagger.type)
+    reindex_boundary = ArchiveConfig.TAGGINGS_COUNT_MIN_CACHE_COUNT
```

Comment:
Should this use a different config variable than the taggings count code? It seems like it might be useful to have them separate so that one can be adjusted without the other.

Reply:
Yeah, that makes sense. I added a new config variable, and the comment should hopefully make it clear that the limit only applies to tagging changes. Changes to the tag itself should reindex no matter what.